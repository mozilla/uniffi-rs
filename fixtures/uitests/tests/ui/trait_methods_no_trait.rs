// Unfortunately, path is relative to a temporary build directory :-/
uniffi_macros::generate_and_include_scaffolding!("../../../../fixtures/trait-methods/src/trait_methods.udl");

fn main() { /* empty main required by `trybuild` */}

// We derive most required traits, just not `Display`, to keep the output smaller.
#[derive(Debug, PartialEq, Eq, Hash)]
pub struct TraitMethods {}

impl TraitMethods {
    fn new(name: String) -> Self {
        unreachable!();
    }
}
