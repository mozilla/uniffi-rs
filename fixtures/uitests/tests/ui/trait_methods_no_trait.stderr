error[E0277]: `TraitMethods` doesn't implement `std::fmt::Display`
 --> $OUT_DIR[uniffi_uitests]/trait_methods.uniffi.rs
  |
  | struct r#TraitMethods { }
  |        ^^^^^^^^^^^^^^ `TraitMethods` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `TraitMethods`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
note: required by a bound in `TraitMethods::uniffi_trait_display::_::{closure#0}::assert_impl_all`
 --> $OUT_DIR[uniffi_uitests]/trait_methods.uniffi.rs
  |
  | #[uniffi::export(Display)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
  = note: this error originates in the macro `::uniffi::deps::static_assertions::assert_impl_all` which comes from the expansion of the attribute macro `uniffi::export` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `TraitMethods` doesn't implement `std::fmt::Display`
 --> $OUT_DIR[uniffi_uitests]/trait_methods.uniffi.rs
  |
  | #[uniffi::export(Display)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^ `TraitMethods` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `TraitMethods`, which is required by `&TraitMethods: std::fmt::Display`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: this error originates in the macro `$crate::__export::format_args` which comes from the expansion of the attribute macro `uniffi::export` (in Nightly builds, run with -Z macro-backtrace for more info)
